polar-core/src/messages.rs

```diff
@@ -36,7 +36,12 @@ impl MessageQueue {
 
     pub fn push(&self, kind: MessageKind, msg: String) {
         let mut messages = self.messages.lock().unwrap();
-        messages.push_back(Message { kind, msg });
+        messages.push_back(Message { kind, msg: msg.clone() });
+
+        // TODO hack to actually get logs.
```

Comment:
This isn't the worst idea, I've needed it as well.

Reply:
Fixed this in a separate PR #431 .